File: todo/src/main/java/kr/jay/todo/api/model/TodoResponse.java
// package kr.jay.todo.api.model;
//
// import java.time.LocalDateTime;
//
// import org.springframework.util.Assert;
//
// import kr.jay.todo.domain.Todo;
// import lombok.Builder;
// import lombok.Data;
//
// @Data
// @Builder
// public class TodoResponse {
//
//     private Long id;
//
//     private String title;
//
//     private String description;
//
//     private Boolean done;
//
//     private LocalDateTime createdAt;
//
//     private LocalDateTime updatedAt;
//
//     public static TodoResponse of(Todo todo) {
//         Assert.notNull(todo, "Todo is null");
//
//         return TodoResponse.builder()
//             .id(todo.getId())
//             .title(todo.getTitle())
//             .description(todo.getDescription())
//             .done(todo.getDone())
//             .createdAt(todo.getCreatedAt())
//             .updatedAt(todo.getUpdatedAt())
//             .build();
//     }
//
// }

File: todo/src/main/java/kr/jay/todo/api/model/TodoResponse.kt
package kr.jay.todo.api.model

import kr.jay.todo.domain.Todo
import java.time.LocalDateTime

/**
 * TodoResponse
 *
 * @author jaypark
 * @version 1.0.0
 * @since 2023/08/17
 */
data class TodoResponse(
    val id: Long,
    val title: String,
    val description: String,
    val done: Boolean,
    val createdAt: LocalDateTime,
    val updatedAt: LocalDateTime?
) {
    companion object {
        fun of(todo: Todo?) : TodoResponse {
            checkNotNull(todo) { "todo must not be null" }
            checkNotNull(todo.id) { "todo Id must not be null" }

            return TodoResponse(
                id = todo.id,
//                id = todo.id!!,
                title = todo.title,
                description = todo.description,
                done = todo.done,
                createdAt = todo.createdAt,
                updatedAt = todo.updatedAt
            )
        }


    }
}